### PR TITLE
Add overflow check when converting large strings to lists columns

### PR DESCRIPTION
**Pull request by davidwendt**
## Description
Fixes a couple places where strings columns are converted to lists column as binary -- chars are represented as INT8.
Since lists columns only support `size_type` offsets type, this change will throw an error if the size of the chars exceeds max `size_type` values.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by davidwendt** [comment]
Perhaps it would be better to check the strings offsets type instead since the type is the real issue regardless of the chars size?
I welcome any opinions here.

**Comment by karthikeyann** [comment]
My suggestion is to check the `char_size`. if a user can't convert large strings, the user may choose to batch it using slices < 2 GB. then offsets type will still remain same but it is possible to perform this operation.
Also, could we move the size check before the column contents are copied?

File: cpp/src/reshape/byte_cast.cu [hunk 1]
```diff
@@ -135,9 +135,14 @@ struct byte_list_conversion_fn<T, std::enable_if_t<std::is_same_v<T, cudf::strin
         input.size(), output_type, stream, mr);
     }
 
-    auto col_content     = std::make_unique<column>(input, stream, mr)->release();
-    auto const num_chars = col_content.data->size();
-    auto uint8_col       = std::make_unique<column>(
+    auto const num_chars = strings_column_view(input).chars_size(stream);
+    CUDF_EXPECTS(num_chars < static_cast<int64_t>(std::numeric_limits<size_type>::max()),
```
Comment (hyperbolic2346) [inline cpp/src/reshape/byte_cast.cu:139]: Why is this one a cast to `int64_t` and the other a cast to `size_t`?

Comment (davidwendt) [reply cpp/src/reshape/byte_cast.cu:139]: Because the type of `num_chars` here is `int64_t`

Comment (mythrocks) [reply cpp/src/reshape/byte_cast.cu:139]: Forgive my ignorance, but I've lost track of why `chars_size` returns an `int64_t` when the number of characters in a `string` column can't exceed `size_type` limits. Wasn't this changed to `int64_t` to allow for super-large string columns?

Comment (davidwendt) [reply cpp/src/reshape/byte_cast.cu:139]: Yes, the chars limit to `size_type` has been removed though this behavior is runtime dependent on an environment variable so as to not break upstream libraries (e.g. Spark).
This has been an ongoing project of work mainly documented here: https://github.com/rapidsai/cudf/issues/13733

**Review by mythrocks — APPROVED** [review]

**Pull request merged**